package NIndexerCore;

import "kernel/indexer/face/blob/markup.proto";

message TDirectText {
    message TDocInfo {
        optional uint32 DocId = 1;
        optional uint64 FeedId = 2;
        optional uint32 UrlFlags = 3;
        optional sint32 Encoding = 4;
        optional uint32 Language = 5;
        optional uint32 Language2 = 6;
        optional sint32 IndexDate = 7;
        optional uint32 MimeType = 8;
        optional string Url = 9;
    }

    message TSpace {
        optional bytes Space = 1;
        optional uint32 BreakType = 2;
    }

    message TToken {
        optional bytes LemmaText = 1;
        optional bytes FormaText = 2;
        optional uint32 Lang = 3;
        optional uint32 Flags = 4;
        optional uint32 Joins = 5;
        optional uint32 FormOffset = 6;
        optional sint32 TermCount = 7;
        optional double Weight = 8;
        optional bytes StemGram = 9;
        optional uint32 FlexGramsCount = 10;
        repeated bytes FlexGrams = 11;
        optional bool IsBastard = 12 [default = false];
        optional uint32 Prefix = 13;
    }

    message TEntry {
        optional fixed32 Posting = 1;
        optional uint32 Offset = 2;
        optional bytes OriginalToken = 3;
        optional uint32 TokensCount = 4;
        optional uint32 SpacesCount = 5;
        repeated TToken Tokens = 6;
        repeated TSpace Spaces = 7;
    }

    message TSentAttr {
        optional uint32 Sent = 1;
        optional bytes Name = 2;
        optional bytes Value = 3;
    }

    message TCounts {
        optional uint64 Entries = 1;
        optional uint64 Zones = 2;
        optional uint64 ZoneAttrs = 3;
        optional uint64 SentAttrs = 4;
    }

    optional TDocInfo DocInfo = 1;
    optional TCounts Counts = 2;
    repeated TEntry Entries = 3;
    repeated TDirectMarkup.TZone Zones = 4;
    repeated TDirectMarkup.TAttribute ZoneAttrs = 5;
    repeated TSentAttr SentAttrs = 6;
}
